### PR TITLE
Add single page application support 

### PR DESCRIPTION
**Pull request by Sipioteo**
Giving the possibility to add a sapFile to use when an endpoint is not found.
Fixed ioutil deprecation.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -92,7 +91,11 @@ func (s3 *S3) Open(name string) (http.File, error) {
 }
 
 func getObject(ctx context.Context, s3 *S3, name string) (*minio.Object, error) {
-	names := [4]string{name, name + "/index.html", name + "/index.htm", "/404.html"}
+	names := []string{name, name + "/index.html", name + "/index.htm"}
+	if sapFile != "" {
+		names = append(names, sapFile)
```
Comment (harshavardhana) [inline main.go:96]: Is sapfile a full path or a relative path ?

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Can you provide examples on how this can be tested ?

File: main.go (hunk 1 above)
Comment (Sipioteo) [reply main.go:96]: > Is sapfile a full path or a relative path ?

Full path, every URL should put always to the same file.

Comment (Sipioteo) [reply main.go:96]: > Is sapfile a full path or a relative path ?

I made a typo. It's not SAP but SPA.
It's Spa = Single Page Application

Comment (harshavardhana) [reply main.go:96]: can you fix the environment variable and also inside the code from `sap -> spa` ?

**Comment by Sipioteo** [comment]
@harshavardhana Done, fixed the typo

**Comment by Sipioteo** [comment]
@harshavardhana if you can merge I can start my SPA server🙂

**Review by harshavardhana — APPROVED** [review]

**Review by harshavardhana — CHANGES_REQUESTED** [review]
[oryxBuildBinary](https://github.com/harshavardhana/s3www/pull/20/files#diff-0cc91448cff3c23248ec636a288566167dae94ade2b69d8aaaa24e9c6a12db6a) 

There is an odd binary attached in the PR can you remove it @Sipioteo

**Comment by Sipioteo** [comment]
@harshavardhana Deleted odd bynary and fixed a typo

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**